### PR TITLE
[Broker] Fix call sync method in async rest api for get tenant namespaces

### PR DESCRIPTION
**Pull request by nodece**
### Motivation

Remove call sync method in async rest api for get tenant namespaces to avoid thread block.

### Modifications

- Cover the sync method to async method

### Documentation

- [x] `no-need-doc` 
  

Signed-off-by: Zixuan Liu <nodeces@gmail.com>

### CODE REVIEW

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/TenantResources.java [hunk 1]
```diff
@@ -110,6 +112,62 @@ public void createTenant(String tenantName, TenantInfo ti) throws MetadataStoreE
...
+
+        // this will return a cluster in v1 and a namespace in v2
+        getChildrenAsync(joinPath(BASE_POLICIES_PATH, tenant)).whenComplete((clusterOrNamespaces, ex) -> {
+            if (ex != null) {
+                result.completeExceptionally(ex);
+                return;
+            }
+
+            List<String> allNamespaces = new ArrayList<>();
+            System.out.println(clusterOrNamespaces);
```
Comment (mattisonchao) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/TenantResources.java:126]: why ``System.out.println(clusterOrNamespaces);``

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/TenantResources.java [hunk 2]
```diff
@@ -110,6 +112,62 @@ public void createTenant(String tenantName, TenantInfo ti) throws MetadataStoreE
         return namespaces;
     }
 
+    public CompletableFuture<List<String>> getListOfNamespacesAsync(String tenant) {
+        CompletableFuture<List<String>> result = new CompletableFuture<>();
```
Comment (mattisonchao) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/TenantResources.java:116]: Could we try to use ``thenAccept`` or  ``thenCompose`` to refactor it?

**Review by mattisonchao — APPROVED** [review]

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/TenantResources.java (hunk 2 above)
Comment (nodece) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/TenantResources.java]: Good catch!

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/TenantResources.java (hunk 1 above)
Comment (nodece) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/TenantResources.java]: bad code.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 3]
```diff
@@ -106,26 +106,32 @@
 
 public abstract class NamespacesBase extends AdminResource {
 
-    protected List<String> internalGetTenantNamespaces(String tenant) {
+    protected void internalGetTenantNamespaces(AsyncResponse asyncResponse, String tenant) {
         checkNotNull(tenant, "Tenant should not be null");
```
Comment (liudezhi2098) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:110]: when Tenant  is  null,  does it need to be asyncResponse

**Pull request closed without merging**